bundles/automation/org.eclipse.smarthome.automation.api/src/main/java/org/eclipse/smarthome/automation/RuleRegistry.java

```diff
@@ -44,10 +46,19 @@
  * </ul>
  *
  * @author Yordan Mihaylov - Initial Contribution
+ * @author Victor Toni - added search capability via {@link Predicate}
  */
 public interface RuleRegistry extends Registry<Rule, String> {
 
     /**
+     * This method is used to get a collection of {@link Rule}s which have the same {@link Predicate}.
+     *
+     * @param predicate to check the rules against
+     * @return collection of {@link Rule}s having matching predicate.
+     */
+    public Collection<Rule> getAll(Predicate<Rule> predicate);
```

Comment:
I'm agree that there should be  a common filtering mechanism,
I think ViToni has choosen to use guava interfaces, because I've said that there was a jdk (i.e. skelmir) which currently does not supported jdk1.8. But you are right the dependence to guava is not a good idea.